src/operator/pad-inl.h

```diff
@@ -73,25 +75,26 @@ class PadOp : public Operator {
     DType constant_value = param_.constant_value;
     // TODO(nswamy@): update the documentation and log below when support is added for more than
     // 4D/5D arrays and not requiring higher dimensions to be zero.
-    if ((rank == 4) && !pad[0] && !pad[1] && !pad[2] && !pad[3]) {
-      Tensor<xpu, 4, DType> data =
-          in_data[pad_enum::kData].get<xpu, 4, DType>(s);
-      Tensor<xpu, 4, DType> out =
-          out_data[pad_enum::kOut].get<xpu, 4, DType>(s);
-      pad_image(out, data, param_.pad_width, param_.mode, constant_value);
-    } else if ((rank == 5) && !pad[0] && !pad[1] && !pad[2] && !pad[3]) {
-      Tensor<xpu, 5, DType> data =
-          in_data[pad_enum::kData].get<xpu, 5, DType>(s);
-      Tensor<xpu, 5, DType> out =
-          out_data[pad_enum::kOut].get<xpu, 5, DType>(s);
-      pad_image(out, data, param_.pad_width, param_.mode, constant_value);
-    } else {
-      LOG(FATAL) << "Current implementation only supports 4D and 5D "
-                    "input arrays with padding applied "
-                    "only on axes 1, 2 and 3. "
-                    "Expects axes 4 and 5 in pad_width to be zero.";
+    switch (rank) {
+      case 4:
+        {
+          Tensor<xpu, 4, DType> data =
+            in_data[pad_enum::kData].get<xpu, 4, DType>(s);
+          Tensor<xpu, 4, DType> out =
+            out_data[pad_enum::kOut].get<xpu, 4, DType>(s);
+          pad_image(out, data, param_.pad_width, param_.mode, constant_value);
+          break;
+        }
+      case 5:
+        {
+          Tensor<xpu, 5, DType> data =
+            in_data[pad_enum::kData].get<xpu, 5, DType>(s);
+          Tensor<xpu, 5, DType> out =
+            out_data[pad_enum::kOut].get<xpu, 5, DType>(s);
+          pad_image(out, data, param_.pad_width, param_.mode, constant_value);
+          break;
+        }
```

Comment:
I think its good practice. If there is a bug in infershape, this would help immediately catch it.